KifuSF/FlaggingViewController.swift

```diff
@@ -0,0 +1,177 @@
+//
+
+import UIKit
+
+class FlaggingViewController: UIViewController {
+    // MARK: - Variables
+    private let flaggingInfoLabel = UILabel(font: UIFont.preferredFont(forTextStyle: .title2),
+                                    textColor: UIColor.Text.SubHeadline)
+    private let flaggingOptionsTableView = UITableView(forAutoLayout: ())
+    
+    var flaggableItems = [FlaggedContentType]()
+    
+    var userToReport: User?
+    var donationToReport: Donation?
+
+    /**
+     <#Lorem ipsum dolor sit amet.#>
+
+     TODO: present loading vc
+     */
+
+    // MARK: - Initializers
+    convenience init(
+        flaggableItems: [FlaggedContentType],
+        userToReport: User? = nil,
+        donationToReport: Donation? = nil) {
+        self.init()
+        
+        self.flaggableItems = flaggableItems
+        self.userToReport = userToReport
+        self.donationToReport = donationToReport
+    }
+
+    // MARK: - Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureDelegates()
+        configureStyling()
+        configureLayout()
+    }
+
+    // MARK: - Methods
+    @objc private func dismissViewController() {
+        dismiss(animated: true)
+    }
+    
+    private func createdReport(report: Report?) {
+        guard report != nil else {
+            UIAlertController(errorMessage: nil)
+                .present(in: self)
+            
+            return
+        }
+        
+        UIAlertController(
+            title: "Thanks for letting us know!",
+            message: "We will shortly review your request and take action if needed.",
+            preferredStyle: .alert)
+            .addCancelButton(title: "Dismiss") { [unowned self] (_) in
+                self.dismissViewController()
+            }
+            .present(in: self)
+    }
+}
+
+// MARK: - UITableViewDataSource
+extension FlaggingViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return flaggableItems.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = UITableViewCell(style: .default, reuseIdentifier: String(describing: flaggingOptionsTableView))
+        
+        cell.layoutMargins = UIEdgeInsets.zero
+        cell.textLabel?.font = UIFont.preferredFont(forTextStyle: .headline)
+        cell.textLabel?.textColor = UIColor.Text.Headline
+        
+        cell.textLabel?.text = flaggableItems[indexPath.row].description
+        
+        return cell
+    }
+}
+
+// MARK: - UITableViewDelegate
+extension FlaggingViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        
+        let selectedItem = flaggableItems[indexPath.row]
+        
+        //TODO: ask the user for a message on why they flagged it
```

Comment:
TODO: this is where the message is added to the report

Reply:
done!

Reply:
Great!